src/audienceManager.ts

```diff
@@ -0,0 +1,98 @@
+import { SDKLoggerApi } from './sdkRuntimeModels';
+import {
+    FetchUploader,
+    XHRUploader,
+    AsyncUploader,
+    fetchPayload
+} from './uploaders';
+import Audience from './audience';
+
+export interface IAudienceMembershipsServerResponse {
+    dt: 'cam';  // current audience memberships
+    ct: number; // timestamp
+    id: string;
+    audience_memberships: Audience[];
+}
+
+export interface IAudienceMemberships {
+    currentAudienceMemberships: Audience[];
+}
+
+export default class AudienceManager {
+    public url: string = '';
+    public userAudienceAPI: AsyncUploader;
+    public logger: SDKLoggerApi;
+
+    constructor(
+        userAudienceUrl: string,
+        apiKey: string,
+        logger: SDKLoggerApi,
+    ) {
+        this.logger = logger;
+        this.url = `https://${userAudienceUrl}${apiKey}/audience`;
+        this.userAudienceAPI = window.fetch
+            ? new FetchUploader(this.url)
+            : new XHRUploader(this.url);
+    }
+
+    public async sendGetUserAudienceRequest(mpid: string, callback: (userAudiences: IAudienceMemberships) => void) {
+        this.logger.verbose('Fetching user audiences from server');
+
+        const fetchPayload: fetchPayload = {
+            method: 'GET',
+            headers: {
+                Accept: '*/*',
+            },
+        };
+        const audienceURLWithMPID = `${this.url}?mpid=${mpid}`;
+        let userAudiencePromise: Response;
+
+        try {
+             userAudiencePromise = await this.userAudienceAPI.upload(
+                 fetchPayload,
+                 audienceURLWithMPID
+             );
+
+            if (
```

Comment:
Is there a reason we're not using a `switch`?

Reply:
I actually did this at first and it wasn't working as expected, according to [this](https://stackoverflow.com/questions/41480811/switch-statement-with-numeric-values), number values don't work in a switch statement, and we'd have to do something like 
```
switch (true) {
   case userAudiencePromise.status >= 200 &&
                userAudiencePromise.status < 300
}
```
I wasn't a fan of that.  However, since http logic is done here and in batchuploader, maybe we can decide what the preferred option is as a follow up and do it to both in a single ticket?